presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoStatement.java

```diff
@@ -103,14 +103,20 @@ public void setMaxFieldSize(int max)
     public int getMaxRows()
             throws SQLException
     {
-        checkOpen();
-        return maxRows.get();
+        return Ints.saturatedCast(getLargeMaxRows());
```

Comment:
I'm not sure this correct. If someone has set something larger than `Integer.MAX_VALUE` with `setLargeMaxRows()`, this should probably throw a `SQLException`. That seems safer than truncating the value (and they had to explicitly set the limit using the other API).